### PR TITLE
Implement sparse system images (fix #299)

### LINKED ISSUE #299: [bullhead][fp2] Package and implement libsparse
Package libsparse from the android project which provides the `img2simg` and `simg2img` commands.
These split a large image for the system partition into seperate smaller chucks with sparse headers.

This is required for at least the Nexus 5x and the Fairphone 2 (#245) because it fixes the `Invalid sparse file format at header magi` error.

https://android.googlesource.com/platform/system/core/+/refs/heads/master/libsparse/

### PR DESCRIPTION
**Pull request by PabloCastellano**
Please test!

Fixes: https://github.com/postmarketOS/pmbootstrap/issues/299

### CODE REVIEW

**Comment by PabloCastellano** [comment]
[Looks like](https://travis-ci.org/PabloCastellano/pmbootstrap/builds/259296400?utm_source=github_status&utm_medium=notification) the tar.gz archive has a different sha512sum every time you download it because the creation date of the files compressed is the exact moment when you download it.

**Comment by PureTryOut** [comment]
Besides the checksum issue, I get the following error:

```
[18:49:29] *** (4/5) FILL INSTALL BLOCKDEVICE ***
[18:49:29] (native) copy rootfs_lg-bullhead to /mnt/install/
[18:49:36] (native) Make sparse system image
[18:49:36] NOTE: The failed command's output is above the ^^^ line in the logfile: /home/bart/.local/var/pmbootstrap/log.txt
[18:49:36] ERROR: Command failed: (native) % cd /home/user/rootfs/ && su user -c 'img2simg lg-bullhead.img lg-bullhead-sparse.img'
[18:49:36] Run 'pmbootstrap log' for details.
[18:49:36] See also: <https://postmarketos.org/troubleshooting>
```

```
(027701) [18:49:36] (native) umount /dev/loop0
(027701) [18:49:36] (native) % losetup -d /dev/loop0
(027701) [18:49:36] (native) Make sparse system image
(027701) [18:49:36] (native) % cd /home/user/rootfs/ && su user -c 'img2simg lg-bullhead.img lg-bullhead-sparse.img'
ash: img2simg: not found
(027701) [18:49:36] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Actually installing the package seems to have gone wrong somewhere...

File: pmb/install/install.py [hunk 1]
```diff
@@ -125,6 +125,16 @@ def install(args):
     fix_mount_folders(args)
     pmb.chroot.shutdown(args, True)
 
+    # Convert system image to sparse using img2simg
+    if args.deviceinfo["flash_sparse"] == "true":
+        logging.info("(native) Make sparse system image")
```
Comment (ollieparanoid) [inline pmb/install/install.py]: Before printing the `(native) make sparse system image` (we've used lowercase letters after the (chroot_suffix) so far), you could *build and install* the program with the following command:
```python
pmb.chroot.apk.install(args, ["pkgname_of_img2simg"])
```

**Comment by PabloCastellano** [comment]
I have rebased this PR from current master but GitHub doesn't seem to reflect all the changes in the web interface (it is ok if you clone the branch)

**Comment by ollieparanoid** [comment]
Great work!

**Pull request merged**